File: ecos-records/src/main/java/ru/citeck/ecos/records3/predicate/model/ComposedPredicate.java
package ru.citeck.ecos.records3.predicate.model;

import ecos.com.fasterxml.jackson210.annotation.JsonProperty;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Objects;

public abstract class ComposedPredicate implements Predicate {

    @JsonProperty("val")
    private List<Predicate> predicates;

    public List<Predicate> getPredicates() {
        return predicates == null ? Collections.emptyList() : predicates;
    }

    public void setPredicates(List<Predicate> predicates) {
        if (predicates == null) {
            this.predicates = new ArrayList<>();
        } else {
            this.predicates = new ArrayList<>(predicates);
        }
    }

    public void addPredicate(Predicate predicate) {
        if (predicates == null) {
            predicates = new ArrayList<>();
        }
        predicates.add(predicate);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        ComposedPredicate that = (ComposedPredicate) o;
        return Objects.equals(predicates, that.predicates);
    }

    @Override
    public int hashCode() {
        return Objects.hash(predicates);
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records2/predicate/model/ComposedPredicate.kt
package ru.citeck.ecos.records2.predicate.model

import com.fasterxml.jackson.annotation.JsonProperty
import kotlin.collections.ArrayList

abstract class ComposedPredicate : Predicate {

    @JsonProperty("val")
    private var predicates: MutableList<Predicate>? = null

    fun getPredicates(): List<Predicate> {
        return predicates ?: arrayListOf()
    }

    fun setPredicates(predicates: List<Predicate>?) {
        this.predicates = ArrayList(predicates ?: emptyList())
    }

    @JsonProperty("v")
    fun setVal(predicates: List<Predicate>?) {
        setPredicates(predicates)
    }

    fun addPredicate(predicate: Predicate) {
        if (predicates == null) {
            predicates = ArrayList()
        }
        predicates!!.add(predicate)
    }
}
